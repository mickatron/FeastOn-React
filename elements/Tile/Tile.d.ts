import * as React from 'react';

export interface TileProps {
  [key: string]: any;
  /** An element type to render as (string or function). */
  as?: any;
  /** Primary content. */
  children?: React.ReactNode;
  /** Additional classes. */
  className?: string | array | object;
  /** BEM modifier attributes.*/
  modifier?: string | array;
  /** Width className */
  width?: string;
}

declare class Tile extends React.Component<TileProps, {}> {
}

export default Tile;