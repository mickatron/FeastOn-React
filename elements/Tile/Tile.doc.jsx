/* eslint-disable */
import React from 'react';
import Component from './Tile';
import PropTypes from 'prop-types';
import asType from '../../proptypes/as';
import classNameType from '../../proptypes/className';

/**
 * A basic tile device often used to group content or implement hierarchy.
 * 
 * ** Sub Components **
 * * `TileHeader` as `Tile.Header`
 * * `TileMeta` as `Tile.Meta`
 * * `TileContent` as `Tile.Content` 
 * * `TileFooter` as `Tile.Footer`
 *
 * @composedWith UiWrapper
*/
function Tile(props) {
  return <Component {...props} />;
}

export default Tile;
