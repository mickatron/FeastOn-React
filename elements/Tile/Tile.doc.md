```js noeditor
{
Tile.Header = TileHeader;
Tile.Content = TileContent;
Tile.Footer = TileFooter;
}
null
```

### Default Tile.
```js
<Tile>
  <Tile.Header>
    <h3>Tile Header</h3>
  </Tile.Header>
  <Tile.Content>
    <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam quis urna ac eros hendrerit convallis. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.</p>

    <p>Proin id ultrices nunc. Cras leo orci, ultrices auctor lacinia sed, maximus a diam. Quisque convallis posuere est, vitae venenatis urna interdum eu.</p>
  </Tile.Content>
  <Tile.Footer>
    <Button>Button</Button>
  </Tile.Footer>
</Tile>
```