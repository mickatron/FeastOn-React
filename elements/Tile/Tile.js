import uiWrapper from '../../HOC/uiWrapper';

export default uiWrapper({
  name: 'Tile',
});
