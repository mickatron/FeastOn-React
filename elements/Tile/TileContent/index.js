import TileContent from './TileContent';

export default TileContent;
