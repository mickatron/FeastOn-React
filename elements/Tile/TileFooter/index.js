import TileFooter from './TileFooter';

export default TileFooter;
