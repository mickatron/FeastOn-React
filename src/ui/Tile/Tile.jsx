import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import classNameType from 'js/shared/proptypes/className';

export default class Tile extends React.PureComponent {
  
  static propTypes = {
    className: classNameType,
    children: PropTypes.node.isRequired,
  };
  
  render() {
    const { className, children } = this.props;
    return <div className={classnames(className,'c-Tile')} >{children}</div>;
  }
}
