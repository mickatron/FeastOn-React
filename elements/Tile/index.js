import Tile from './Tile';
import TileHeader from './TileHeader';
import TileMeta from './TileMeta';
import TileContent from './TileContent';
import TileFooter from './TileFooter';

Tile.Header = TileHeader;
Tile.Meta = TileMeta;
Tile.Content = TileContent;
Tile.Footer = TileFooter;

export default Tile;
