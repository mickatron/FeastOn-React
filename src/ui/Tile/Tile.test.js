import { shallow } from 'enzyme';
import React from 'react';
import Tile from './Tile';

describe('Tile', () => {

  let component = null;

  beforeAll(() => {
    component = shallow(
      <Tile 
        className="test"
      >
        <p>Tile Content</p>
      </Tile>
    );
    expect(component).toMatchSnapshot();
  });

  describe('renders', ()=>{
    it('has custom className', ()=>{
      expect(component.find('.c-Tile.test').exists()).toBeTruthy();
    });

    it('has children', () => { 
      expect(component.find('.c-Tile p').exists()).toBeTruthy();
    });

  });
});