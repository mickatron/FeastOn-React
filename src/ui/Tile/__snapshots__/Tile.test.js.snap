// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="test c-Tile"
>
  <p>
    Tile Content
  </p>
</div>
`;
